blog/_posts/2021-08-20-symbiosis-symbulation.md

```diff
@@ -0,0 +1,83 @@
+
+```cpp
+void AvidaGPTournamentSelect(AvidaGPWorld & world, size_t t_size, size_t tourny_count=1) {
+    emp_assert(t_size > 0, "Cannot have a tournament with zero organisms.", t_size, world.GetNumOrgs());
+    emp_assert(t_size <= world.GetNumOrgs(), "Tournament too big for world.", t_size, world.GetNumOrgs());
+    emp_assert(tourny_count > 0);
+
+    emp::vector<size_t> entries;
+    for (size_t T = 0; T < tourny_count; T++) {
+      entries.resize(0);
+      // Choose organisms for this tournament (with replacement!)
+      for (size_t i=0; i < t_size; i++) entries.push_back(
+        world.GetRandomOrgID()
+      );
+
+      double best_fit = world.CalcFitnessID(entries[0]);
+      size_t best_id = entries[0];
+
+      // Search for a higher fit org in the tournament.
+      for (size_t i = 1; i < t_size; i++) {
+        const double cur_fit = world.CalcFitnessID(entries[i]);
+        if (cur_fit > best_fit) {
+          best_fit = cur_fit;
+          best_id = entries[i];
+        }
+      }
+
+      // Place the highest fitness into the next generation!
+      world.DoBirth(world.GetOrgPtr(best_id), best_id, 1 );
+    }
+  }
+```
+
+
+
+ ## Problems Encountered
+During this summer, I had a lot of internet and power issues which caused delays with the overall progression of this project. Initially we had tried to override the empirical world birth format, known as DoBirth, however there isn't any way to get the original host organism from 'mem' since it's just the genome and not the whole organism. As a result, the elite select and tournament select functions were recreated to fit our DoBirth method for this project.
+
+ ## Conclusion
+ I had a great time working on this project despite the delays due to having a multitude of power and internet issues. I've gained valuable skills, such as working with others, and analyzing files. On top of this, the enrichment seminars were really helpful with gaining new perspectives and skills that I will carry into my career. WAVES is a great intersection between computer science and biology, something that was very interesting and also tough for me due to my lack of knowledge in the biology area. My advice for the next onboarding student is to communicate well with your mentors and team when you feel overwhelmed! Empirical is such a large library and while very interesting, most files will not be necessary for your own project. 
```

Comment:
Really like this section! It's cool that you gave an overview of the program as a whole. And the advice for your predecessor is a great idea!
Might copy you on that 😜.